application-urlshortener-ui/src/main/resources/URLShortener/WebHome.xml

```diff
@@ -36,5 +36,285 @@
   <minorEdit>false</minorEdit>
   <syntaxId>xwiki/2.1</syntaxId>
   <hidden>true</hidden>
-  <content/>
+  <content>{{warning}}
+{{translation key="urlshortener.duplicates.warning.slow"/}}
+{{/warning}}
+
+----
+
+= {{translation key="urlshortener.duplicates.title"/}} =
+
+{{velocity output="false"}}
+#template('hierarchy_macros.vm')
+## Convenience method.
+#macro (escape $str)
+  $escapetool.xml($services.rendering.escape($str, 'xwiki/2.1'))
+#end
+{{/velocity}}
+
+{{velocity wiki="true"}}
+#set ($q = $services.query.createQuery('property.URLShortener.Code.URLShortenerClass.pageID:*',
+    'solr').setLimit(2000).execute().get(0).getResults())
+## Group shortURL pages by pageID.
+#set ($map = {})
+#foreach ($i in $q)
+  #set ($shortURLs = $i.get('property.URLShortener.Code.URLShortenerClass.pageID_string'))
+  #set ($ref = $services.solr.resolveDocument($i))
+  #set ($ref = $ref.withoutLocale())
+  #foreach ($shortURL in $shortURLs)
+    #if (!$map.keySet().contains($shortURL))
+      #set ($discard = $map.put($shortURL, [$ref]))
+    #elseif (!$map.get($shortURL).contains($ref))
+      #set ($discard = $map.get($shortURL).add($ref))
+    #end
+  #end
+#end
+
+{{html wiki=true}}
+#set ($noDuplicates = true)
+#foreach ($shortUrl in $map.keySet())
+  #if ($map.get($shortUrl).size() &gt; 1)
+    #set ($noDuplicates = false)
+    == {{translation key="urlshortener.duplicates.title.section" parameters="$shortUrl,$map.get($shortUrl).size()"/}} ==
+    #foreach ($foundDoc in $map.get($shortUrl))
+      |&lt;button class='btn urlshortener-regenerate-btn' data-old-page-id='#escape($shortUrl)' data-page-ref='#escape($foundDoc)'&gt;{{translation key="urlshortener.duplicates.regenerate.button"/}}&lt;/button&gt;|(%style="width: 100%;"%)$services.rendering.escape("#hierarchy($foundDoc, {})", 'xwiki/2.1')|
```

Comment:
You are missing the use of `translation` macro, with the `$services.rendering.escape`. You should only use `$services.rendering.escape`, in order to not need the `wiki=true`
Also, this line is longer than 120characters, you need some new lines